core/system/src/mpu/vmpu.c

```diff
@@ -449,11 +449,15 @@ static int copy_box_namespace(const char *src, char *dst)
 
         if (src[bytes_copied] == '\0') {
             /* We've reached the end of the box namespace. */
-            break;
+            return bytes_copied + 1 /* Plus 1 for the terminating-null */;
         }
     }
 
-    return bytes_copied;
+    /* We did not find a terminating null in the src. The src has been verified
+     * in vmpu_box_namespace_from_id as being in the box config table. It is a
+     * programmer error if the namespace in the box config table is not
+     * null-terminated, so we halt. */
+    HALT_ERROR(SANITY_CHECK_FAILED, "box namespace missing terminating-null");
```

Comment:
I think the comment got lost with a rebase. Could you capitalize and punctuate the halt error message?